test/e2e/apimachinery/webhook.go

```diff
@@ -828,7 +828,7 @@ var _ = SIGDescribe("AdmissionWebhook [Privileged:ClusterAdmin]", func() {
 		_, err := createValidatingWebhookConfiguration(ctx, f, validatingWebhookConfiguration)
 		gomega.Expect(err).To(gomega.HaveOccurred(), "create validatingwebhookconfiguration should have been denied by the api-server")
 		expectedErrMsg := "compilation failed"
-		gomega.Expect(strings.Contains(err.Error(), expectedErrMsg)).To(gomega.BeTrue())
+		gomega.Expect(strings.Contains(err.Error(), expectedErrMsg)).To(gomega.BeTrueBecause("expected error to contain: %q, got: %q", expectedErrMsg, err.Error()))
```

Comment:
Instead of checking the boolean result of some Go expression, this and several of the checks below should be replaced by the corresponding Gomega check. In this case:

```suggestion
		gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring(expectedErrMsg)))
```

In general, a message which dumps the content of what is being checked is suspicious. Dumping that is the job of the test helper.

Reply:
I found two match in webhook.go and updated them. Let me know if I missed any other cases.